controller/src/main/java/org/jboss/as/controller/logging/ControllerLogger.java

```diff
@@ -3629,4 +3629,7 @@ OperationFailedRuntimeException capabilityAlreadyRegisteredInContext(String capa
 
     @Message(id = 476, value = "Value for attribute '%s' is invalid: either '%s' must be specified on its own or '%s' needs to be specified with at least one of '%s' or '%s'")
     OperationFailedException invalidCredentialReferenceValue(String attributeName, String clearTextAttributeName, String credentialStoreAttributeName, String clearTextAttributeName1, String alias);
+
+    @Message(id = 477, value = "Parameter name '%s' is invalid.")
+    OperationFailedException invalidParameterName(String parameterName);
```

Comment:
The return type here should be IllegalArgumentException, as that is what you intend to throw.